internal/services/eventgrid/eventgrid_domain_resource.go

```diff
@@ -369,6 +400,12 @@ func resourceEventGridDomainRead(d *pluginsdk.ResourceData, meta interface{}) er
 		if props := model.Properties; props != nil {
 			d.Set("endpoint", props.Endpoint)
 
+			dataResidencyBoundary := string(domains.DataResidencyBoundaryWithinGeopair)
+			if props.DataResidencyBoundary != nil {
+				dataResidencyBoundary = string(*props.DataResidencyBoundary)
+			}
+			d.Set("data_residency_boundary", dataResidencyBoundary)
```

Comment:
`pointer.From` can do the nil check for us
```suggestion
			d.Set("data_residency_boundary", pointer.From(props.DataResidencyBoundary))
```